In [ ]:
RANDOM FORREST was the technique usd for the model training.

# Task
Prepare features (`X`) from the `X_pca_df` and target (`y`) from the 'Risk Category' column in the `df` DataFrame. Split these into training and testing sets (80/20 ratio) with a `random_state` for reproducibility, and then summarize the sizes of the resulting datasets.

## Prepare Features and Target

### Subtask:
Define the feature set `X` using `X_pca_df` (the PCA-transformed data) and the target variable `y` as the 'Risk Category' from the original DataFrame `df`.


**Reasoning**:
To define the feature set `X` and target variable `y` as per the subtask, I will assign the `X_pca_df` to `X` and extract the 'Risk Category' column from `df` to `y`.



In [ ]:
import pandas as pd

# 1. Assign the X_pca_df DataFrame to a new variable named X
X = X_pca_df

# 2. Extract the 'Risk Category' column from the df DataFrame and assign it to a new variable named y
y = df['Risk Category']

print("Feature set X and target variable y prepared successfully!")

print("\nFirst five rows of X (PCA-transformed features):")
display(X.head())

print("\nFirst five rows of y (Risk Category target variable):")
display(y.head())

Feature set X and target variable y prepared successfully!

First five rows of X (PCA-transformed features):


,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10,PC11,PC12,PC13,PC14,PC15,PC16,PC17,PC18,PC19,PC20
0,-0.212262,-0.065884,1.230684,0.923961,-0.069382,-0.385573,-0.567068,0.292565,-0.530442,-0.162424,-1.451268,-0.251565,0.448461,-0.178792,0.707215,0.177246,0.161652,-0.019297,-0.222697,-0.267124
1,-0.456115,-0.090471,-1.372292,-0.505112,-0.141229,-0.990020,0.231355,1.118413,0.514906,-0.213972,-1.157197,-0.372710,-1.175375,0.492046,0.709281,0.131280,0.258358,-0.000440,-0.229972,-0.351794
2,-0.228996,-0.002352,-0.973515,1.001796,-0.021643,-0.694810,-0.497039,0.184643,-0.296550,0.096399,-1.390710,0.034120,0.727720,0.105195,0.696503,-0.264722,0.667636,0.236320,-0.024145,0.000197
3,0.927498,0.170807,-1.152626,0.783617,0.225111,-0.640933,-0.486204,0.128099,-0.565649,-0.164507,-1.144621,0.468843,0.472797,0.097292,-0.041203,0.356557,-0.115216,-0.080738,0.073306,-0.059477
4,-0.238044,-0.123921,-0.674909,0.514036,-0.651781,0.045263,0.319115,-0.399142,0.315399,0.308674,0.558528,0.173216,0.390798,0.150806,-0.145129,0.542188,0.634071,0.036442,0.072404,0.202447



First five rows of y (Risk Category target variable):


,Risk Category
0,Good
1,Good
2,Good
3,Good
4,Good


**Reasoning**:
To split the feature set `X` and target variable `y` into training and testing sets with an 80/20 ratio, I will use the `train_test_split` function from `sklearn.model_selection`. I will also set a `random_state` for reproducibility and then display the shapes of the resulting datasets to summarize their sizes.



In [ ]:
from sklearn.model_selection import train_test_split

# Split the data into training and testing sets (80/20 ratio)
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

print("Data split into training and testing sets successfully!")

# Summarize the sizes of the resulting datasets
print("\nShape of X_train:", X_train.shape)
print("Shape of X_test:", X_test.shape)
print("Shape of y_train:", y_train.shape)
print("Shape of y_test:", y_test.shape)


Data split into training and testing sets successfully!

Shape of X_train: (815, 20)
Shape of X_test: (204, 20)
Shape of y_train: (815,)
Shape of y_test: (204,)


### Summary of Dataset Split

The data has been successfully split into training and testing sets with an 80/20 ratio.

**Training Set Sizes:**
*   `X_train` (features): 815 samples with 20 principal components.
*   `y_train` (target): 815 samples.

**Testing Set Sizes:**
*   `X_test` (features): 204 samples with 20 principal components.
*   `y_test` (target): 204 samples.

This split ensures that the model will be trained on 80% of the data and evaluated on the remaining 20%, providing a reliable assessment of its generalization performance.

## Final Task

### Subtask:
Provide a comprehensive summary of the data splitting process and its importance for machine learning model development.


## Summary:

### Data Analysis Key Findings

*   The feature set `X` was successfully prepared using the PCA-transformed data (`X_pca_df`), consisting of 20 principal components.
*   The target variable `y` was correctly extracted as the 'Risk Category' from the original DataFrame `df`.
*   The data was split into training and testing sets using an 80/20 ratio with `random_state=42` for reproducibility.
*   The resulting dataset sizes are:
    *   `X_train` (features): 815 samples with 20 principal components.
    *   `X_test` (features): 204 samples with 20 principal components.
    *   `y_train` (target): 815 samples.
    *   `y_test` (target): 204 samples.

### Insights or Next Steps

*   The 80/20 train-test split with a fixed random state ensures a robust and reproducible framework for training machine learning models and evaluating their performance on unseen data.
*   The next logical step is to train a suitable machine learning model (e.g., a classification algorithm) using the `X_train` and `y_train` datasets, and then evaluate its performance on the `X_test` and `y_test` sets to assess its generalization capabilities.


# Task
Train a RandomForestClassifier from `sklearn.ensemble` using `X_train` and `y_train`, with `random_state=42`.

## Train Random Forest Classifier

### Subtask:
Instantiate a RandomForestClassifier from `sklearn.ensemble` with `random_state=42` for reproducibility and train it using the `X_train` and `y_train` datasets.


**Reasoning**:
To instantiate and train the RandomForestClassifier, I will import it from `sklearn.ensemble`, create an instance with the specified `random_state`, and then fit it to the training data `X_train` and `y_train`.



In [ ]:
from sklearn.ensemble import RandomForestClassifier

# Instantiate RandomForestClassifier with random_state=42
rf_classifier = RandomForestClassifier(random_state=42)

# Train the classifier using the X_train and y_train datasets
rf_classifier.fit(X_train, y_train)

print("Random Forest Classifier trained successfully!")

Random Forest Classifier trained successfully!


### Summary of Random Forest Classifier Training

The `RandomForestClassifier` was successfully instantiated with `random_state=42` to ensure reproducibility. The model was then trained using the `X_train` (PCA-transformed features) and `y_train` (Risk Category labels) datasets.

This marks the successful completion of the training phase, and the `rf_classifier` model is now ready for making predictions and evaluating its performance on the unseen test set.

## Final Task

### Subtask:
Summarize that the Random Forest Classifier has been trained.


## Summary:

### Data Analysis Key Findings
*   A `RandomForestClassifier` was successfully instantiated with `random_state=42` to ensure reproducibility.
*   The model was trained using the provided `X_train` and `y_train` datasets.
*   The training process concluded successfully, as confirmed by the message: "Random Forest Classifier trained successfully!".

### Insights or Next Steps
*   The trained `rf_classifier` model is now prepared to make predictions and can be used for evaluating its performance on unseen data.
*   The next step should involve evaluating the model's performance using appropriate metrics on a test dataset.


# Task
Make predictions using the trained `rf_classifier` on the `X_test` dataset.

## Make Predictions

### Subtask:
Use the trained `rf_classifier` to make predictions on the `X_test` dataset.


**Reasoning**:
To make predictions as per the subtask, I will use the `.predict()` method of the trained `rf_classifier` on the `X_test` dataset and store the results in `y_pred`.



In [ ]:
y_pred = rf_classifier.predict(X_test)
print("Predictions made on the test set successfully!")


Predictions made on the test set successfully!
